runtime/util/hshelp.c

```diff
@@ -482,6 +485,180 @@ areSingleSlotConstantRefsIdentical(J9ROMConstantPoolItem * romCP1, U_32 index1,
+			sigIndex += 1;
+		}
+	}
+	/* for an object this will move past the ;, for a primitive this will move past the argument */
+	sigIndex += 1;
+	return sigIndex;
+}
+
+static UDATA
+areCallSiteDataMethodsEquivalent(J9ROMClass* romClass1, UDATA callSiteIndex1, J9ROMClass* romClass2, UDATA callSiteIndex2)
+{
+	/**
+	 * Structure of CallSiteData structure in ROM class (also see romclasswalk.c)
+	 * NAS:
+	 * 		callSiteCount SRP to J9ROMNameAndSignature structures describing the resolved callsite. Note: SRP is 32 bits
+	 * bsm indices:
+	 * 		callSiteCount 16 bit indices into bsmData. for example 0 is the first bsm, 1 is the second etc.
+	 * bsm data:
+	 * 		(header) The first 16 bits contains the index of the bsm's MethodHandle in the constant pool.
+	 * 		(header) The second 16 bits contains the number of bsm arguments in addition to the required three (MethodHandles.Lookup, String, MethodType).
+	 * 		The additional BSM arguments, these are constant pool indices.
+	 */
+	J9ROMConstantPoolItem *romCP1 = J9_ROM_CP_FROM_ROM_CLASS(romClass1);
+	J9ROMConstantPoolItem *romCP2 = J9_ROM_CP_FROM_ROM_CLASS(romClass2);
+	J9SRP *callSiteData1 = (J9SRP *) J9ROMCLASS_CALLSITEDATA(romClass1);
+	J9SRP *callSiteData2 = (J9SRP *) J9ROMCLASS_CALLSITEDATA(romClass2);
+	/* get call site name and signature */
+	J9ROMNameAndSignature* nas1 = SRP_PTR_GET(callSiteData1 + callSiteIndex1, J9ROMNameAndSignature*);
+	J9ROMNameAndSignature* nas2 = SRP_PTR_GET(callSiteData2 + callSiteIndex2, J9ROMNameAndSignature*);
+	/* get call site BSM index */
+	U_16 *bsmIndices1 = (U_16 *) (callSiteData1 + romClass1->callSiteCount);
+	U_16 *bsmIndices2 = (U_16 *) (callSiteData2 + romClass2->callSiteCount);
+	U_16 bsmIndex1 = bsmIndices1[callSiteIndex1];
+	U_16 bsmIndex2 = bsmIndices2[callSiteIndex2];
+	/* get top of bsm data */
+	U_16 *bsmData1 = findBSMDataAtIndex(bsmIndices1 + romClass1->callSiteCount, bsmIndex1);
+	U_16 *bsmData2 = findBSMDataAtIndex(bsmIndices2 + romClass2->callSiteCount, bsmIndex2);
+	/* additional arg variables */
+	U_16 additionalArgCount1 = bsmData1[1];
+	U_16 additionalArgCount2 = bsmData2[1];
+	/* BSM MethodHandle reference */
+	J9ROMMethodHandleRef *mhRef1 = (J9ROMMethodHandleRef*) &romCP1[bsmData1[0]];
+	J9ROMMethodHandleRef *mhRef2 = (J9ROMMethodHandleRef*) &romCP2[bsmData2[0]];
+	U_32 bsmHandleTypeAndCpType1 = mhRef1->handleTypeAndCpType;
+	U_32 bsmHandleTypeAndCpType2 = mhRef2->handleTypeAndCpType;
+	U_32 bsmCpTypeIndex1 = bsmHandleTypeAndCpType1 & J9DescriptionCpTypeMask;
+	U_32 bsmCpTypeIndex2 = bsmHandleTypeAndCpType2 & J9DescriptionCpTypeMask;
+	UDATA bsmCpType1 = J9_CP_TYPE(J9ROMCLASS_CPSHAPEDESCRIPTION(romClass1), bsmCpTypeIndex1);
+	UDATA bsmCpType2 = J9_CP_TYPE(J9ROMCLASS_CPSHAPEDESCRIPTION(romClass2), bsmCpTypeIndex2);
+	/* additional helpers */
+	J9ROMMethodRef* bsmMethod = NULL;
+	J9ROMNameAndSignature *bsmNAS = NULL;
+	J9UTF8 *bsmSig = NULL;
+	U_8* sigData = NULL;
+	U_32 sigIndex = 0;
+	U_32 sigLength = -1;
```

Comment:
Why -1? It's not valid for U_32, and the variable appears to never be read before the real value is assigned. 0 is probably a better default.